src/Api/Serializer/BasicUserSerializer.php

```diff
@@ -45,6 +45,10 @@ protected function getDefaultAttributes($user)
      */
     protected function groups($user)
     {
-        return $this->hasMany($user, GroupSerializer::class);
+        if ($this->getActor()->can('viewHiddenGroups')) {
+            return $this->hasMany($user, GroupSerializer::class);
+        }
+
+        return $this->hasMany($user, GroupSerializer::class, 'visibleGroups');
```

Comment:
I haven't seen this approach in other serializers. Do we have a way to use `whereVisibleTo()` here as well? :thinking:

Reply:
I don't think so: we're returning a tobscure Relationship object, not a query.